clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -49,6 +49,14 @@ RValue CIRGenFunction::emitBuiltinExpr(const GlobalDecl &gd, unsigned builtinID,
     }
   }
 
+  switch (builtinID) {
+  case Builtin::BI__builtin_complex: {
+    mlir::Value real = emitScalarExpr(e->getArg(0));
+    mlir::Value imag = emitScalarExpr(e->getArg(1));
+    return RValue::getComplex(real, imag);
```

Comment:
Following the suggestion by @Lancern below, we should be calling createComplexCreate here rather than returning separate values.

Reply:
Yes, once #144519 is merged, I will update it